app/views/courses/TeacherClassView.coffee

```diff
@@ -424,6 +446,9 @@ module.exports = class TeacherClassView extends RootView
   onChangeCourseSelect: (e) ->
     @trigger 'course-select:change', { selectedCourse: @courses.get($(e.currentTarget).val()) }
 
+  onChangeLockedLevelSelect: (e) ->
+    @trigger 'locked-level-select:change', { selectedLevel: $(e.currentTarget).val() }
```

Comment:
@wjllance @AndrewJakubowicz why are we using a trigger event instead of calling `@setSelectedCourseLockedLevel` directly?

Reply:
> @wjllance @AndrewJakubowicz why are we using a trigger event instead of calling `@setSelectedCourseLockedLevel` directly?

emm, actually I used to call @setSelectedCourseLockedLevel directly, and it works fine. While I just happened to notice we use event trigger for Course Select, I didn't figure out why, so I just copied the logic...

Reply:
Gotcha.  I think these Backbone events are usually used when disconnected components need to send messages to each other, but in this case it's within the same component so can just call directly.  Backbone events are quite powerful but a bit dangerous in how the open up the possibility of side effects or complex dependencies so we should try to avoid them unless they're necessary.